### PR TITLE
filter out invalid cluster names

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
prevent any client send invalid clusterNames

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v2/ads.go [hunk 1]
```diff
@@ -558,16 +558,23 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 						continue
 					}
 				}
+				validClusters := []string{}
+				for _, cn := range clusters {
+					_, _, hostname, _ := model.ParseSubsetKey(cn)
+					if len(hostname) > 0 {
```
[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply pilot/pkg/proxy/envoy/v2/ads.go]: Isn't there any inbound clusters?

[the remark this answers is not in the document]
Comment (rshriram) [reply pilot/pkg/proxy/envoy/v2/ads.go]: CDS sends outbound clusters only

[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply pilot/pkg/proxy/envoy/v2/ads.go]: Learn that, thanks.

**Pull request closed without merging**